src/mon/MDSMonitor.cc

```diff
@@ -738,7 +738,7 @@ bool MDSMonitor::prepare_beacon(MonOpRequestRef op)
         info->state_seq = seq;
       });
 
-      if (state == MDSMap::STATE_ACTIVE) {
+      if (info.state != MDSMap::STATE_ACTIVE && state == MDSMap::STATE_ACTIVE) {
```

Comment:
`info` is a reference that will have been modified in place by `modify_daemon`, so I don't think this will ever be true.  Probably can just swap this block with the preceding one, or be even more explicit by adding an `old_state` temporarily for this check?

Reply:
Hmm, now I'm not sure how this solution actually worked for me (probably my testing was not right). I'll fix this...